interpreter/interpreter.go

```diff
@@ -1465,3 +1466,53 @@ func (a *arguments) listUnused() []string {
 	}
 	return unused
 }
+
+type contextKey int
+
+const (
+	callStackKey contextKey = iota
+)
+
+// StackEntry describes a single entry in the call stack.
+type StackEntry struct {
+	FunctionName string
+	Location     ast.SourceLocation
+}
+
+// stackElement describes a single entry in the call stack.
+type stackElement struct {
+	parent *stackElement
+	entry  StackEntry
+	depth  int
+}
+
+// Stack retrieves the call stack for a given context.
+func Stack(ctx context.Context) []StackEntry {
```

Comment:
Do you think we should add another parameter to the function interface? I wanted to avoid that because of the breaking change but it was the other way I was thinking of doing this.

Reply:
Would it break `influxdb` or `idpe`? Or just the public interface between Flux packages?

Reply:
It could potentially break influxdb or idpe. It's a public interface for functions so anywhere that invokes a flux function through Go code would be affected. I'm not sure if it is invoked that way. I am trying to search.

One potential positive for putting it on the context though is this data could be considered request scoped data. It is context information for this specific call. I don't think context has to be http request-scoped data but just scoped data to a specific call. At the same time, it might be good to have it as part of an explicit struct that is part of the function call.

Reply:
> One potential positive for putting it on the context though is this data could be considered request scoped data. It is context information for this specific call.

Ok, since you put it like that, I'm okay with deferring to your judgement on this.